include/aspect/particle/world.h

```diff
@@ -21,729 +21,310 @@
-        {
-          dof_handler = new_dh;
-        };
+        void advect_particles();
 
         /**
-         * Get the deal.II BlockVector solution associated with this particle
-         * world.
-         *
-         * @return The BlockVector solution for this world.
+         * Access to particles in this world.
          */
-        const LinearAlgebra::BlockVector *get_solution() const
-        {
-          return solution;
-        };
+        std::multimap<LevelInd, Particle<dim> > &
+        get_particles();
 
         /**
-         * Set the deal.II BlockVector solution associated with this particle
-         * world.
-         *
-         * @param [in] new_solution The new LinearAlgebra::BlockVector
-         * solution for this world.
+         * Const access to particles in this world.
          */
-        void set_solution(const LinearAlgebra::BlockVector *new_solution)
-        {
-          solution = new_solution;
-        };
+        const std::multimap<LevelInd, Particle<dim> > &
+        get_particles() const;
 
         /**
-         * Set the particle Integrator scheme for this particle world.
-         *
-         * @param [in] new_integrator The new Integrator scheme for this
-         * world.
+         * Advance particles by the old timestep using the current
+         * integration scheme. This accounts for the fact that the tracers
+         * are actually still at their old positions and the current timestep
+         * length is already updated for the next step at the time this
+         * function is called.
          */
-        void set_integrator(Integrator::Interface<dim, T> *new_integrator)
-        {
-          integrator = new_integrator;
-        };
+        void advance_timestep();
 
         /**
-         * Set the MPI communicator for this world.
+         * Calculates the global sum of particles over all processes. This is
+         * done to monitor the number of particles that have fallen out of the
+         * domain.
          *
-         * @param [in] new_comm_world The new MPI_Comm object for this world.
+         * @return Total number of particles in simulation.
          */
-        void set_mpi_comm(const MPI_Comm new_comm_world)
-        {
-          communicator = new_comm_world;
-
-          // Determine the size of the MPI comm world
-          world_size = Utilities::MPI::n_mpi_processes(communicator);
-          self_rank  = Utilities::MPI::this_mpi_process(communicator);
-        };
+        unsigned int get_global_particle_count() const;
 
         /**
-         * Get the MPI communicator associated with this particle world.
-         *
-         * @return associated MPI_Comm
+         * This callback function is called by Simulator to allow us to connect
+         * to the SimulatorSignals.
          */
-        MPI_Comm mpi_comm()
-        {
-          return communicator;
-        };
+        void
+        connector_function(aspect::SimulatorSignals<dim> &signals);
```

Comment:
I don't understand from the description or its name what this function does. Can you clarify both?